File: template/springboot/TinyPro/src/main/java/com/TinyPro/entity/po/Role.java
package com.TinyPro.entity.po;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serializable;
import java.util.List;

@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "role")
public class Role implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @TableId(type = IdType.AUTO)
    private Integer id;

    @Column(nullable = false, unique = true) // 角色名称唯一且非空
    private String name;

    @ManyToMany(cascade = {CascadeType.PERSIST, CascadeType.MERGE})
    @JoinTable(
            name = "role_permission",
            joinColumns = @JoinColumn(name = "roleId"), // 改为蛇形命名
            inverseJoinColumns = @JoinColumn(name = "permissionId")
    )
    private List<Permission> permission;

    @ManyToMany(cascade = {CascadeType.PERSIST, CascadeType.MERGE})
    @JoinTable(
            name = "role_menu",
            joinColumns = @JoinColumn(name = "roleId"), // 改为蛇形命名
            inverseJoinColumns = @JoinColumn(name = "menuId")
    )
    private List<Menu> menus;

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public List<Permission> getPermission() {
        return permission;
    }

    public void setPermission(List<Permission> permission) {
        this.permission = permission;
    }

    public List<Menu> getMenus() {
        return menus;
    }

    public void setMenus(List<Menu> menus) {
        this.menus = menus;
    }
}

File: template/nestJs/libs/models/src/role.ts
import {
  Column,
  Entity,
  JoinTable,
  ManyToMany,
  PrimaryGeneratedColumn,
} from 'typeorm';
import { Permission } from './permission';
import { Menu } from './menu';
import { ApiProperty } from '@nestjs/swagger';

@Entity('role')
export class Role {
  @ApiProperty({
    description: '角色ID'
  })
  @PrimaryGeneratedColumn()
  id: number;
  @ApiProperty({
    description: '角色名称 (人类可读)'
  })
  @Column()
  name: string;

  @ApiProperty({
    description: '角色所拥有的权限',
    type: [Permission]
  })
  @ManyToMany(() => Permission, {
    onUpdate: 'CASCADE',
  })
  @JoinTable({ name: 'role_permission' })
  permission: Permission[];

  @ApiProperty({
    description: '角色所拥有的菜单',
    type: [Menu]
  })
  @ManyToMany(() => Menu)
  @JoinTable({ name: 'role_menu' })
  menus: Menu[];
}
